richtextfx/src/main/java/org/fxmisc/richtext/model/SegmentOps.java

```diff
@@ -68,15 +68,19 @@ public String getText(Either<L, R> seg) {
     }
 
     @Override
-    public Either<L, R> subSequence(Either<L, R> seg, int start, int end) {
-        return seg.map(l -> lOps.subSequence(l, start, end),
-                       r -> rOps.subSequence(r, start, end));
+    public Optional<Either<L, R>> subSequence(Either<L, R> seg, int start, int end) {
+        return seg.unify(ll -> seg.unify(l -> lOps.subSequence(l, start, end).map(Either::left),
+                                         e -> Optional.empty()),
+                         rr -> seg.unify(e -> Optional.empty(),
+                                         r -> rOps.subSequence(r, start, end).map(Either::right)));
```

Comment:
Getting a `Type mismatch: cannot convert from Object to Optional<Either<L,R>>` with that - since `subSequence()` now returns an `Optional<>`, don't I have to consider `lOps/rOps.subSequence()` being empty, similar to what `SegmentOps.join()` operation already does:

    public Optional<Either<L, R>> join(Either<L, R> left, Either<L, R> right) {
        return left.unify(ll -> right.unify(rl -> lOps.join(ll, rl).map(Either::left), 
                                                           rr -> Optional.empty()),
                                  lr -> right.unify(rl -> Optional.empty(), 
                                                           rr -> rOps.join(lr, rr).map(Either::right)));
    }

Still getting started with reactfx, so sorry if that is a trivial question :smile: